.buildkite/ci.mjs

```diff
@@ -532,30 +563,41 @@ function getBuildBunStep(platform, options) {
  * @returns {Step}
  */
 function getTestBunStep(platform, options, testOptions = {}) {
-  const { os } = platform;
+  const { asan, os } = platform;
   const { buildId, unifiedTests, testFiles } = testOptions;
 
+  // Use the proper target key for the step name
   const args = [`--step=${getTargetKey(platform)}-build-bun`];
   if (buildId) {
     args.push(`--build-id=${buildId}`);
   }
   if (testFiles) {
     args.push(...testFiles.map(testFile => `--include=${testFile}`));
   }
-
   const depends = [];
   if (!buildId) {
     depends.push(`${getTargetKey(platform)}-build-bun`);
   }
 
+  let label = `${getPlatformLabel(platform)} - test-bun`;
+  let key = `${getPlatformKey(platform)}`;
+  if (asan) {
+    label += " (ASAN)";
+    key += "-asan";
+  }
+
+  key += "-test-bun";
+
   return {
-    key: `${getPlatformKey(platform)}-test-bun`,
-    label: `${getPlatformLabel(platform)} - test-bun`,
+    key,
+    label,
     depends_on: depends,
     agents: getTestAgent(platform, options),
     retry: getRetry(),
     cancel_on_build_failing: isMergeQueue(),
-    parallelism: unifiedTests ? undefined : os === "darwin" ? 2 : 10,
+    // Use reduced parallelism for ASAN builds to avoid running out of memory
```

Comment:
This comment doesn't make any sense? The parallelism here is for tests, not builds. Reducing this will make the build take longer?